pkg/rpcclient/nep24/royalty.go

```diff
@@ -0,0 +1,182 @@
+// Package nep24 provides RPC wrappers for NEP-24 contracts, which define NFT royalties.
+package nep24
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/nspcc-dev/neo-go/pkg/neorpc/result"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/nep11"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/neptoken"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+)
+
+// RoyaltyRecipient contains information about the recipient and the royalty amount.
+type RoyaltyRecipient struct {
+	Address util.Uint160
+	Amount  *big.Int
+}
+
+// RoyaltiesTransferredEvent represents a RoyaltiesTransferred event as defined in the NEP-24 standard.
+type RoyaltiesTransferredEvent struct {
+	RoyaltyToken     util.Uint160
+	RoyaltyRecipient util.Uint160
+	Buyer            util.Uint160
+	TokenID          []byte
+	Amount           *big.Int
+}
+
+// RoyaltyReader is an interface for contracts implementing NEP-24 royalties.
+type RoyaltyReader struct {
+	neptoken.Base
+	Invoker nep11.Invoker
```

Comment:
It should not work like this, because the base for NEP24 is either NEP11 divisible or NEP11 non-divisible. I'd suggest:
1. Remove `neptoken.Base` from `RoyaltyReader`.
2. Replace `nep11.Invoker` with `Invoker`.
3. Adjust RPC binding in the following way: if contract implements NEP-11, then only structures from the `nep11` package to the contract reader. If contract implements both NEP-11 and NEP-24, then include both `nep11` and `nep24` structures.